src/Sarif.Viewer.VisualStudio/Controls/InfoBar.cs

```diff
@@ -19,6 +20,14 @@ namespace Microsoft.Sarif.Viewer.Controls
     /// </summary>
     public class InfoBar : IVsInfoBarUIEvents
     {
+        /// <summary>
+        /// Gets a list of the active info bar instances created by this class.
+        /// </summary>
+        /// <remarks>
+        /// Exposed for testing purposes.
+        /// </remarks>
+        public static List<InfoBar> InfoBars { get; } = new List<InfoBar>();
```

Comment:
>public [](start = 8, length = 6)

The fact that this is public and is public from the assembly means that anyone else (not just our test code) can easily get access to these.

What exactly are we going to do with this in the tests?

Reply:
I no longer have this member. Instead I now have `Shown` and `Closed` events that both product code and test code can subscribe to. See the test PR for usage.

---
In reply to: [521657580](https://github.com/microsoft/sarif-visualstudio-extension/pull/268#discussion_r521657580) [](ancestors = 521657580)

Reply:
It is now internal. I tried making it public to overcome a problem that turned out to be pilot error.

---
In reply to: [521657580](https://github.com/microsoft/sarif-visualstudio-extension/pull/268#discussion_r521657580) [](ancestors = 521657580)